### PR TITLE
Add tf iac

### PR DESCRIPTION
**Pull request by jdbass**
## Description of the Change

Adds Tofu/Terragrunt IaC and github actions workflows to manage release & deployment

## Benefits

- defines a consistent Appointment infrastructure for all stages of the SDLC
- provides pipelines to manage build, testing & deployment of infrastructure & code

### CODE REVIEW

File: .github/workflows/deploy-production.yml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        working-directory: ./tofu/environments/stage/services/backend-service
+        run: |
+          terragrunt init -upgrade
+          terragrunt validate
+          terragrunt plan -var "image=$(cat steps.get_ecr_tag.outputs.*)" -out tfplan
+          cat tfplan
+#          terragrunt apply tfplan
+
+#      - name: Deploy frontend to S3
+#        run: aws s3 sync ./frontend/frontend/dist "s3://${{ needs.deploy-iac.outputs.bucket }}"
```
[the remark this answers is not in the document]
Comment (jdbass) [reply .github/workflows/deploy-production.yml:162]: This is just disabled temporarily so we can verify the production release workflow without actually overwriting the production environment

File: .github/workflows/deploy-production.yml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+          terragrunt plan -out tfplan
+#          terragrunt apply tfplan
+          
+      - name: frontend-infra
+        working-directory: ./tofu/environments/stage/services/frontend-infra
+        run: |
+          terragrunt init -upgrade
+          terragrunt validate
+          terragrunt plan -out tfplan
+#          terragrunt apply tfplan
```
[the remark this answers is not in the document]
Comment (jdbass) [reply .github/workflows/deploy-production.yml:97]: These are just disabled temporarily so we can verify the production release workflow without actually overwriting the production environment

File: .github/workflows/deploy-production.yml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        id: output-cloudfront-distro
+        working-directory: ./tofu/environments/stage/services/frontend-infra
+        run: |
+          output=$(terragrunt output cloudfront_id)
+          echo cloudfront_id=$output >> $GITHUB_OUTPUT
+
+  release-production:
+    name: Release to Production
+    needs: deploy-iac
+    if: startsWith(github.ref_name, 'r-') # the prefix we have added to the tag
```
[the remark this answers is not in the document]
Comment (jdbass) [reply .github/workflows/deploy-production.yml:116]: tag name, it's created by the create-release job in the deploy-staging workflow.  We can change how it's named if desired

[the remark this answers is not in the document]
Comment (MelissaAutumn) [reply .github/workflows/deploy-production.yml:116]: Naw it's good, thanks!

**Review by MelissaAutumn — APPROVED** [review]

**Pull request merged**